saw-remote-api/src/SAWServer/ProofScript.hs

```diff
@@ -61,13 +79,28 @@ instance FromJSON Prover where
   parseJSON =
     withObject "prover" $ \o -> do
       (name :: String) <- o .: "name"
+      let unints = fromMaybe [] <$> o .:? "uninterpreted functions"
       case name of
-        "abc"   -> pure ABC
-        "cvc4"  -> CVC4 <$> o .: "uninterpreted functions"
-        "rme"   -> pure RME
-        "yices" -> Yices <$> o .: "uninterpreted functions"
-        "z3"    -> Z3 <$> o .: "uninterpreted functions"
-        _       -> empty
+        "abc"            -> pure ABC_Internal
+        "internal-abc"   -> pure ABC_Internal
+        "rme"            -> pure RME
+        "sbv-abc"        -> pure SBV_ABC_SMTLib
+        "sbv-boolector"  -> SBV_Boolector <$> unints
+        "sbv-cvc4"       -> SBV_CVC4  <$> unints
+        "cvc4"           -> SBV_CVC4  <$> unints
+        "sbv-mathsat"    -> SBV_MathSAT <$> unints
+        "mathsat"        -> SBV_MathSAT <$> unints
+        "sbv-yices"      -> SBV_Yices <$> unints
+        "yices"          -> SBV_Yices <$> unints
+        "sbv-z3"         -> SBV_Z3    <$> unints
```

Comment:
I'm not sure I quite understand the conventions used for determining which things have distinguished names in this list. For example, there's both `sbv-z3` and `z3` (which are the same thing internally), but while there's `sbv-boolector`, there's no `boolector`.

Perhaps what I really want is documentation about this in `old-SAW.rst`—or better yet, autogenerated documentation in `SAW.rst` :)

Reply:
Ah, maybe there should be a plain `boolector`. The idea is that you can explicitly pick the backend you want, if desired, or choose the un-prefixed name which will give you whichever the developers have decided is the default. This mirrors the convention in SAWScript where we adopted it for backward compatibility. In the remote API, that's less critical, but I guess it can't hurt to be consistent with SAWScript?

I'll look into `old-SAW.rst` and see whether there's anything I could add there.